### PR TITLE
Testing inserts on CockroachDB

### LINKED ISSUE #507: CockroachDB support
Sqlx has a known issue with transations and MARS
https://github.com/launchbadge/sqlx/issues/933

Does sea orm has a workaround to do multiple inserts inside single transaction with CockroachDB?

tokio-postgres has no problems, so it might be fixed on client side somehow without waiting for https://github.com/cockroachdb/cockroach/issues/40195

### PR DESCRIPTION
**Pull request by billy1624**
## PR Info

- Closes #507

## Summary

Try SeaORM inserts on CockroachDB.

Tracking issue: cockroachdb/cockroach#40195

Could workaround it as suggested on launchbadge/sqlx#933, we can use `fetch_all` instead of `fetch_one` as a temporary solution. However, this impose additional performance penalty for other databases that don't require this workaround.

### CODE REVIEW

File: src/executor/insert.rs [hunk 1]
```diff
@@ -118,7 +118,8 @@ where
             let cols = PrimaryKey::<A>::iter()
                 .map(|col| col.to_string())
                 .collect::<Vec<_>>();
-            let res = db.query_one(statement).await?.unwrap();
+            let vec = db.query_all(statement).await?;
+            let res = vec.get(0).unwrap();
```
Comment (billy1624) [inline src/executor/insert.rs:122]: Avoid `fetch_one`

File: src/executor/insert.rs [hunk 2]
```diff
@@ -163,19 +164,20 @@ where
...
+            .all(db)
             .await?
         }
         false => {
             let insert_res =
                 exec_insert::<A, _>(primary_key, db_backend.build(&insert_statement), db).await?;
             <A::Entity as EntityTrait>::find_by_id(insert_res.last_insert_id)
-                .one(db)
+                .limit(1)
+                .all(db)
```
Comment (billy1624) [inline src/executor/insert.rs:175]: And, here

**Review by billy1624 — COMMENTED** [review]
Hey @sergeyshaykhullin, I get it work by changing these lines in SeaORM core.

**Comment by sergeyshaykhullin** [comment]
Does this fixes can be a part of 0.7 release? Should it be a separate feature like "postgres", but "cockroachdb"?

**Comment by billy1624** [comment]
> Does this fixes can be a part of 0.7 release? Should it be a separate feature like "postgres", but "cockroachdb"?

I think we could have release it in 0.7.0. However, we need to write CI tests to make sure it works on a live CockroachDB.

@tyt2y3 thoughts?

**Comment by tyt2y3** [comment]
Yes. I think we definitely need automated tests setup before releasing.
I think the best approach in the long term is to treat CockroachDB as a separate backend from Postgres.

**Comment by billy1624** [comment]
Added CockroachDB tests and `cockroachdb` feature

CC @tyt2y3

**Comment by ducc** [comment]
Hey @billy1624 I've just tested this by crudely copy pasting the src/executor/insert.rs changes into my own fork (https://github.com/duccltd/sea-orm/commit/23f81f462c6574614099f1736602baa9bb4b0870).

My problem was that I was unable to insert a row, however these changes fix it for me!

**Comment by billy1624** [comment]
Hey @ducc, thanks for testing it!

**Pull request closed without merging**